### DISCUSSION TITLE
SEO Image Metadata For Articles

### DISCUSSION THREAD

**Discussion by Umayarz18** [carlson-technologies/coffeeclass.io#76] [Ideas]
When writing articles, is there a spot to place image metadata? I notice if I try to share an article on twitter it does not provide a preview the same way other sites do.

For better viewability on social platforms, would you be willing to add SEO images for articles? I can work on that but just wanted to see if that's something you'd be open to having for articles.

**Comment by bjcarlson42** [comment]
Yeah this is definitely something we could use! At the moment, the default image is our logo with a white background. The set for articles can be found here: https://github.com/carlson-technologies/coffeeclass.io/blob/main/src/components/SEO.js

**Reply by Umayarz18** [reply]
Sweet, I'll take a look when I get the time

**Reply by Umayarz18** [reply]
Started playing with the SEO component on my machine, did a couple of changes but I'm not sure how to check if it is working locally. Let me know what else I should add to it and I can make a PR.

### Updated `SEO.js`
![image](https://user-images.githubusercontent.com/61628982/151885296-81f73c58-eaf3-424f-9732-1655fa5b325e.png)

### Updated Example Article
![image](https://user-images.githubusercontent.com/61628982/151886926-6a9e98d8-21a9-47b8-bc5a-52b616cdf87a.png)

^^ Think this format will work for writers if they want a unique OG:image
### Changes
- Added Tags to `NextSEO` object.
- Added image prop to `SEO.js` component
  - changed `featuredImage.url` to be either the default or the provided image prop
- Updated ArticleJsonLD images from `featuredImage` to `featuredImage.url` based on how the next-seo docs have it.
  -  Docs have it like this 🔽 
![image](https://user-images.githubusercontent.com/61628982/151884860-6610022a-46fe-41cc-9575-0bd3faa01f02.png)

**Reply by bjcarlson42** [reply]
1. Looks good! To check locally, you should be able to inspect element and expand the head. From there you can check all the tags. See example image.
<img width="494" alt="Screen Shot 2022-02-01 at 12 37 40 PM" src="https://user-images.githubusercontent.com/48105099/152020770-757a2d01-ca28-4e1e-895d-b1eea1f49410.png">
 
2. Also, we should be able to clean up the frontMatter. Remember DRY (Don't repeat yourself). instead of having `content/articles/<article-name>/imag.jpg` lets just have `image.jpg`. The `content/articles` is always the same and the folder path can be fetched via `useRouter`.

3. Thanks for adding the tags! Great addition.

**Reply by Umayarz18** [reply]
Will add those changes!

Currently looks like this when you put images. I'm about to make a PR on it and I'll let you take a look to see if it works properly.
![image](https://user-images.githubusercontent.com/61628982/152064577-fe9fc309-ea18-4547-b71e-618a3dd3ef60.png)

**Comment by bjcarlson42** [comment]
Hey @Umayarz18 For the go image, take a look at this I just added to the roadmap: https://benjamincarlson.notion.site/As-a-user-I-want-a-proper-image-to-be-shown-when-I-share-an-article-to-facebook-twitter-etc-so-th-a64e348e29d1497e94b028200b99389c

**Reply by bjcarlson42** [reply]
Also, everything you had in the old PR was good (tags, etc). Just don't require a feature image

**Reply by bjcarlson42** [reply]
Also, I have updated the contributions guide for this website: https://benjamincarlson.notion.site/Contributing-to-Coffeeclass-io-27ab5e894368424a9c86a7f11555514b I hope it is even more clear! You'll see at the bottom that I linked the article you wrote!
